bin/viewprof.hs

```diff
@@ -85,9 +86,15 @@ usage = do
 parseArgs :: IO Args
 parseArgs = do
   args <- getArgs
-  if length args /= 1
-     then usage
-     else return $ Args (head args)
+  case args of
+    [arg] -> validateArgs (Args arg)
+    _     -> usage
+ where
+ validateArgs :: Args -> IO Args
+ validateArgs args =
+    do b <- doesFileExist (profilePath args)
+       when (not b) usage
+       pure args
```

Comment:
Could you reformat the code a little like below?

* No vertical alignment around `->` in the case expression
* 2 space before `where`
* 4 space before local definitions under `where`
* Move `do` next to `=`